enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+      ]
+    }
+    ```
+
+    Trust relationship policy
+    ```shell
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Principal": {
+            "AWS": "arn:aws:iam::<managed-cluster-account-id>:role/ocm-managed-cluster-<md5-hash-unique-identifier>"
+          },
+          "Action": "sts:AssumeRole",
+          "Condition": {  
+            "StringEquals": {
+                    "aws:PrincipalTag/managed_cluster_name":"<managed-cluster-name>"
+                    "aws:PrincipalTag/hub_cluster_name":"<hub-cluster-name>"
+                    "aws:PrincipalTag/hub_account_id":"<hub-account-id>"
+            }
+          }
+        }
+      ]
+    }
+    ```
+- Creates an entry in the `aws-auth` configmap for the newly created IAM role and maps it to a group with following name. The `ManagedCluster` gets permissions inside the hub through `system:open-cluster-management:<clusterName>` group.
+- Updates `spec.hubAcceptsClient` on `ManagedCluster` to `true`
+
+4. hub-controller creates a namespace as the name of managed cluster on hub cluster if it does not exist.
+- managed cluster can only join a hub once, and it can join to multiple hubs.
+- The UID of the managed cluster is identical on each of the hub the Klusterlet agent joins.
+ 
+5. hub-controller creates a clusterrolebinding on the hub, binds role with the identity of
+   `open-cluster-management:managedcluster:<clusterName>` to group `system:open-cluster-management:<clusterName>`
+     - Allows status update of `ManagedCluster`
+
+6. hub-controller creates rolebinding `open-cluster-management:managedcluster:<clusterName>:registration` bound to cluster role `open-cluster-management:managedcluster:registration` on the cluster namespace on the hub
+- Allows the access of agent on managed cluster to the namespace.
+
+7. hub-controller creates rolebinding `open-cluster-management:managedcluster:<clusterName>:work` bound to cluster role `open-cluster-management:managedcluster:work` on the cluster namespace on the hub
+- Allows the access of agent on managed cluster to pull `ManifestWork` from the namespace.
+
+8. hub-controller updates condition of `ManagedCluster` to `HubAcceptedManagedCluster`.
+
+9. agent on managed cluster assumes its role on hub, and create a new kubeconfig `hub-kubeconfig-secret` using following command and saves it as secret:
+    ```shell
+    aws eks update-kubeconfig --name <hub-eks-cluster-name> --kubeconfig <path/to/create/kubeconfig> --role-arn <role-on-hub-for-managed>
+    ```
+
+10. agent on managed cluster connects to hub apiserver using the new kubeconfig.
+
+11. agent on managed cluster updates conditions of `ManagedCluster` as `ManagedClusterJoined`.
+
+12. agent on managed cluster appends updates other fields in status of `ManagedCluster`. 
+
+13. agent on managed cluster pulls the`ManifestWork` and creates resource.
+
+Registration workflow
+![OCM-using-AWS-IAM.jpg](OCM-using-AWS-IAM.jpg)
+
```

Comment:
maybe we should also write an unjoin flow. Or if accept is updated false, will controller remove iam roles also?

Reply:
We will have to enhance hub controller to remove iam roles, as today it [removes](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/registration/hub/managedcluster/controller.go#L212) k8s rbac resources only. However setting accept to false wont remove ManagedCluster CR, can "setting accept to false" be updated to remove ManagedCluster CR instead of just letting it stay there with [reason denied](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/registration/hub/managedcluster/controller.go#L139)?

Reply:
I think we should remove iam role upon setting accept to false OR manageCluster CR is deleted.

Reply:
ManagedCluster doesn't have permission to delete ManagedCluster CR, so the only option we have is to update hubAcceptsClient field to false. Today if we set hubAcceptsClient field to false, it deletes all resources on hub but not the managedcluster CR itself. I hope it is ok if we enhance ManagedCluster controller on hub to delete ManagedCluster CR at the end, as we want to cleanup everything managedcluster specific resource on hub, when spoke triggers unjoin.

Reply:
Yes, we will remove the IAM resources on hub when spoke admin sets spec.hubAcceptsClient to false on ManagedCluster CR on hub.